python/pyarrow/io.pxi

```diff
@@ -121,6 +121,14 @@ cdef class NativeFile(_Weakrefable):
     def __exit__(self, exc_type, exc_value, tb):
         self.close()
 
+    def __repr__(self):
+        name = f"{self.__class__.__module__}.{self.__class__.__name__}"
```

Comment:
In general those objects are exposed in the top-level `pyarrow` namespace, so I would maybe hardcode that here instead of using `__module__` which gives `pyarrow.lib` (the lib submodule is also considered private)

Reply:
https://github.com/apache/arrow/pull/13921/commits/febd2efa417feba35b57ca41481b18c0992fd9c8